File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp

class Solution {
public:
    int maxLevelSum(TreeNode* root) {
         queue<TreeNode*> q;
        q.push(root);
        int level=1, sum=root->val, l=1;
        while(!q.empty()) {
            int size=q.size(), cur=0;
            for(int i=0;i<size;i++) {
                TreeNode* tmp=q.front();
                 q.pop();
                cur+=tmp->val;
                if(tmp->left!=NULL) q.push(tmp->left);
                if(tmp->right!=NULL) q.push(tmp->right);
            }
            if(cur>sum) {
                sum=cur;
                level=l;
            }
            l++;
        }
        return level;
    }
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.java

class Solution {
    public int maxLevelSum(TreeNode root) {
     Queue<TreeNode> q=new LinkedList<>();
        q.offer(root);
        int level=1, sum=root.val, l=1;
        while(!q.isEmpty()) {
            int size=q.size(), cur=0;
            for(int i=0;i<size;i++) {
                TreeNode tmp=q.poll();
                cur+=tmp.val;
                if(tmp.left!=null) q.offer(tmp.left);
                if(tmp.right!=null) q.offer(tmp.right);
            }
            if(cur>sum) {
                sum=cur;
                level=l;
            }
            l++;
        }
        return level;    
    }
}